### PR TITLE
Adding yala support

### PR DESCRIPTION
**Pull request by macartur**

### CODE REVIEW

File: pyof/foundation/basic_types.py [hunk 1]
```diff
@@ -238,7 +245,7 @@ class IPAddress(GenericType):
     max_prefix = UBInt32(32)
 
     def __init__(self, address="0.0.0.0/32"):
-        """The constructor takes the parameters below.
+        """Create a IPAddress with the parameters below.
```
Comment (cemsbr) [inline pyof/foundation/basic_types.py]: a**n** IPAddress

File: pyof/foundation/basic_types.py [hunk 2]
```diff
@@ -531,6 +552,7 @@ def pack(self, value=None):
             msg = "{} pack error: {}".format(type(self).__name__, err)
             raise exceptions.PackException(msg)
 
+    # pylint: disable-msg=arguments-differ
```
Comment (cemsbr) [inline pyof/foundation/basic_types.py]: This should not affect the rest of the class. Either write in the same lame as `def` (recommended) or enable it right after the method.

File: pyof/foundation/basic_types.py [hunk 3]
```diff
@@ -627,13 +651,15 @@ def insert(self, index, item):
...
                 type than the one specified in the constructor.
+
         """
         if issubclass(item.__class__, self._pyof_class):
             list.insert(self, index, item)
         else:
             raise exceptions.WrongListItemType(item.__class__.__name__,
                                                self._pyof_class.__name__)
 
+    # pylint: disable-msg=arguments-differ
```
Comment (cemsbr) [inline pyof/foundation/basic_types.py]: This should not affect the rest of the class. Either write in the same lame as `def` (recommended) or enable it right after the method.

File: pyof/foundation/network_types.py [hunk 4]
```diff
@@ -35,7 +35,7 @@ class ARP(GenericStruct):
     def __init__(self, htype=1, ptype=0x800, hlen=6, plen=4, oper=1,
                  sha='00:00:00:00:00:00', spa='0.0.0.0',
                  tha="00:00:00:00:00:00", tpa='0.0.0.0'):
-        """The constructor receives the parameters below.
+        """Create a ARP with the parameters below.
```
Comment (cemsbr) [inline pyof/foundation/network_types.py]: a**n** ARP

File: pyof/foundation/network_types.py [hunk 5]
```diff
@@ -198,8 +201,8 @@ class Ethernet(GenericStruct):
     ether_type = UBInt16()
     data = BinaryData()
 
-    def __init__(self, destination=None, source=None, vlan=VLAN(),
-                 ether_type=None, data=b''):
+    def __init__(self, destination=None,
+                 source=None, vlan=VLAN(), ether_type=None, data=b''):
```
Comment (cemsbr) [inline pyof/foundation/network_types.py]: Is this change needed?

File: pyof/foundation/network_types.py [hunk 6]
```diff
@@ -332,7 +341,7 @@ def pack(self, value=None):
                                                        type(self).__name__)
             raise PackException(msg)
 
-    def unpack(self, buffer, offset=0):
+    def unpack(self, buff, offset=0):
```
Comment (cemsbr) [inline pyof/foundation/network_types.py]: Why?

File: pyof/foundation/network_types.py [hunk 7]
```diff
@@ -404,11 +415,11 @@ class IPv4(GenericStruct):
     #: data (:class:`BinaryData`): Packet data
     data = BinaryData()
 
-    def __init__(self, version=4, ihl=5, dscp=0, ecn=0, length=0, # noqa
+    def __init__(self, version=4, ihl=5, dscp=0, ecn=0, length=0,
                  identification=0, flags=0, offset=0, ttl=255, protocol=0,
                  checksum=0, source="0.0.0.0", destination="0.0.0.0",
                  options=b'', data=b''):
-        """The constructor receives the parameters below.
+        """Instanciate a IPv4 with the parameters below.
```
Comment (cemsbr) [inline pyof/foundation/network_types.py]: a**n** IPv4

File: pyof/v0x01/common/action.py [hunk 8]
```diff
@@ -65,7 +65,7 @@ class ActionHeader(GenericStruct):
     _allowed_types = ()
 
     def __init__(self, action_type=None, length=None):
-        """The following constructor parameters are optional.
+        """Create a ActionHeader with the optional parameters below.
```
Comment (cemsbr) [inline pyof/v0x01/common/action.py]: a**n** ActionHeader

File: pyof/v0x01/common/action.py [hunk 9]
```diff
@@ -121,7 +122,7 @@ class ActionOutput(ActionHeader):
     _allowed_types = ActionType.OFPAT_OUTPUT,
 
     def __init__(self, port=None, max_length=UBINT16_MAX_VALUE):
-        """The following constructor parameters are optional.
+        """Create a ActionOutput with the optional parameters below.
```
Comment (cemsbr) [inline pyof/v0x01/common/action.py]: a**n** ActionOutput

File: pyof/v0x01/common/action.py [hunk 10]
```diff
@@ -170,7 +171,7 @@ class ActionEnqueue(ActionHeader):
     _allowed_types = ActionType.OFPAT_ENQUEUE,
 
     def __init__(self, port=None, queue_id=None):
-        """The following constructor parameters are optional.
+        """Create a ActionEnqueue with the optional parameters below.
```
Comment (cemsbr) [inline pyof/v0x01/common/action.py]: a**n**

File: pyof/v0x01/common/action.py [hunk 11]
```diff
@@ -196,7 +197,7 @@ class ActionVlanVid(ActionHeader):
     _allowed_types = ActionType.OFPAT_SET_VLAN_VID,
 
     def __init__(self, vlan_id=None):
-        """The following constructor parameters are optional.
+        """Create a ActionVlanVid with the optional parameters below.
```
Comment (cemsbr) [inline pyof/v0x01/common/action.py]: a**n** (I'll stop here, but there's lots of them).

File: pyof/v0x01/symmetric/echo_reply.py [hunk 12]
```diff
@@ -23,7 +23,7 @@ class EchoReply(GenericMessage):
     data = BinaryData()
 
     def __init__(self, xid=None, data=None):
-        """The constructor takes the parameters below.
+        """Create a EchoReply with the optional parameters below.
```
Comment (cemsbr) [inline pyof/v0x01/symmetric/echo_reply.py]: a**n** Echo...

File: pyof/v0x04/common/flow_match.py [hunk 13]
```diff
@@ -192,7 +192,7 @@ class OxmTLV(GenericStruct):
 
     def __init__(self, oxm_class=None, oxm_field=None,
                  oxm_hasmask=False, oxm_value=None):
-        """Inialize OXM TLV struct.
+        """Create a OXM TLV struct with the optional parameters below.
```
Comment (cemsbr) [inline pyof/v0x04/common/flow_match.py]: a**n**

File: pyof/v0x04/controller2switch/common.py [hunk 14]
```diff
@@ -527,7 +526,7 @@ class OxmProperty(Property):
 
     def __init__(self, property_type=TableFeaturePropType.OFPTFPT_MATCH,
                  oxm_ids=ListOfOxmHeader()):
-        """Constructor of OxmProperty receives the parameters bellow.
+        """Create a OxmProperty with the optional parameters below.
```
Comment (cemsbr) [inline pyof/v0x04/controller2switch/common.py]: a**n** O...

File: setup.py [hunk 15]
```diff
@@ -5,7 +5,8 @@
 """
 from abc import abstractmethod
 # Disabling checks due to https://github.com/PyCQA/pylint/issues/73
-from distutils.command.clean import clean  # pylint: disable=E0401,E0611
+# pylint: disable=import-error,no-name-in-module
```
Comment (cemsbr) [inline setup.py]: This should not affect the rest of the module. Either write in the same lame as `from` (recommended) or enable it right after the import.

File: pyof/v0x01/common/action.py (hunk 11 above)
Comment (renanrodrigo) [reply pyof/v0x01/common/action.py]: there are*
:p

File: pyof/foundation/network_types.py (hunk 6 above)
Comment (macartur) [reply pyof/foundation/network_types.py]: This method in super class has the name buff not buffer.

File: pyof/foundation/basic_types.py (hunk 2 above)
Comment (diraol) [reply pyof/foundation/basic_types.py]: @cemsbr `same lame` == `same line` ?

File: setup.py (hunk 15 above)
Comment (macartur) [reply setup.py]: FIxed.

**Comment by beraldoleal** [comment]
@macartur can you please update this branch because of conflicts. Sorry about that.

@cemsbr please review this and let me if I can accept this.

**Comment by macartur** [comment]
This MR was updated.

File: setup.py [hunk 16]
```diff
@@ -124,10 +126,9 @@ def run(self):
               'sphinx-autobuild',
               'sphinx-rtd-theme',
               'sphinx_bootstrap_theme~=0.4.0',
-              'pydocstyle~=1.1.1',
-              'pylama~=7.3.3',
-              'pylama_pylint~=3.0.1',
-              'radon~=1.5.0'
+              'pydocstyle',
```
Comment (diraol) [inline setup.py]: @macartur we do not need `pydocstyle` (neither `radon`) since they are already required by yala.

File: setup.py (hunk 15 above)
Comment (cemsbr) [reply setup.py]: @macartur, the way it is, in a line of its own and with no indentation, these errors are disabled from line 8 to the end of the file. They should be disabled only in one line.

**Comment by cemsbr** [comment]
Thank you, @macartur. This issue required many changes.
